llvm/lib/Target/AArch64/GISel/AArch64PostSelectOptimize.cpp

```diff
@@ -107,6 +108,8 @@ bool AArch64PostSelectOptimize::doPeepholeOpts(MachineBasicBlock &MBB) {
   for (auto &MI : make_early_inc_range(make_range(MBB.begin(), MBB.end()))) {
     Changed |= foldSimpleCrossClassCopies(MI);
   }
+  for (auto &MI : make_early_inc_range(make_range(MBB.begin(), MBB.end())))
```

Comment:
Unfortunately not. The problem is, if `foldSimpleCrossClassCopies` triggers, then `MI` gets invalidated because it gets deleted and calling `foldCopyDup` then causes a crash.

Reply:
It is possible to check the return value of `foldSimpleCrossClassCopies` and call `foldCopyDup` based on that, however that can cause a situation where we miss the new instruction that `foldSimpleCrossClassCopies` generates.

Reply:
We should try to avoid iterating over the instructions multiple times wherever possible. We might, in the long run, want to have this loop switch on the opcode. Having it continue if foldSimpleCrossClassCopies returns true is Ok for the time being though.

Reply:
Okay, that works. Awesome.